node/src/components/network/conman.rs

```diff
@@ -0,0 +1,1054 @@
+    /// Shared configuration settings.
+    cfg: Config,
+    /// Callback handler for connection setup and incoming request handling.
+    protocol_handler: Box<dyn ProtocolHandler>,
+    /// Juliet RPC configuration.
+    rpc_builder: RpcBuilder<{ super::Channel::COUNT }>,
+    /// The shared state.
+    state: RwLock<ConManState>,
+    /// Our own address (for loopback filtering).
+    public_addr: SocketAddr,
+    /// Our own node ID.
+    our_id: NodeId,
+    /// Limiter for incoming connections.
+    incoming_limiter: Arc<Semaphore>,
+}
+
+/// Share state for [`ConMan`].
+///
+/// Tracks outgoing and incoming connections.
+#[derive(Debug, Default)]
+pub(crate) struct ConManState {
+    /// A set of outgoing address for which a handler is currently running.
+    address_book: HashSet<SocketAddr>,
+    /// Mapping of [`SocketAddr`]s to an instant in the future until which they must not be dialed.
+    do_not_call: HashMap<SocketAddr, Instant>,
+    /// The current route per node ID.
+    ///
+    /// An entry in this table indicates an established connection to a peer. Every entry in this
+    /// table is controlled by an `OutgoingHandler`, all other access should be read-only.
+    routing_table: HashMap<NodeId, Route>,
+    /// A mapping of `NodeId`s to details about their bans.
+    banlist: HashMap<NodeId, Sentence>,
+}
+
+impl ConManState {
+    /// Returns a reference to the address book of this [`ConManState`].
+    pub(crate) fn address_book(&self) -> &HashSet<SocketAddr> {
+        &self.address_book
+    }
+
+    /// Returns a reference to the do not call of this [`ConManState`].
+    pub(crate) fn do_not_call(&self) -> &HashMap<SocketAddr, Instant> {
+        &self.do_not_call
+    }
+
+    /// Returns a reference to the routing table of this [`ConManState`].
+    pub(crate) fn routing_table(&self) -> &HashMap<NodeId, Route> {
+        &self.routing_table
+    }
+
+    /// Returns a reference to the banlist of this [`ConManState`].
+    pub(crate) fn banlist(&self) -> &HashMap<NodeId, Sentence> {
+        &self.banlist
+    }
+}
+
+/// Record of punishment for a peers malicious behavior.
+#[derive(Debug)]
+pub(crate) struct Sentence {
+    /// Time until the ban is lifted.
+    pub(crate) until: Instant,
+    /// Justification for the ban.
+    pub(crate) justification: BlocklistJustification,
+}
+
+/// Data related to an established connection.
+#[derive(Debug)]
+pub(crate) struct Route {
+    /// Node ID of the peer.
+    // TODO: Consider removing this, as it is already represented in the key.
+    pub(crate) peer: NodeId,
+    /// The established [`juliet`] RPC client that is used to send requests to the peer.
+    pub(crate) client: RpcClient,
+    /// The remote address of the peer.
+    ///
+    /// For outgoing connections, this will be the peer address we connected to, for incoming ones
+    /// it is the usually randomly selected outgoing address of the peer.
+    pub(crate) remote_addr: SocketAddr,
```

Comment:
There's an important distinction here, a `peer` ID is "self-chosen" or fixed. The same is true for the `peer_addr`, since it is the public address that peer listens on.

`remote_addr` can either be the `peer_addr`, or it can be a randomly generated address in case the peer initiated the connection.